cookiecutter/generate.py

```diff
@@ -250,8 +253,7 @@ def generate_file(
 
     logger.debug('Writing contents to file %s', outfile)
 
-    with open(outfile, 'w', encoding='utf-8', newline=newline) as fh:
-        fh.write(rendered_file)
+    Path(outfile).write_text(rendered_file, encoding='utf-8', newline=newline)
```

Comment:
Wow, ruff really boned this line. CI fails on Pythons 3.9 and lower.

This will need to be addressed.

Reply:
done

Reply:
this is now fixed upstream in ruff - https://github.com/astral-sh/ruff/issues/12222

that shouldn't stop this PR from getting merged, but the error suppression will need to be removed next time ruff is updated